include/FeedingFanOut.h

```diff
@@ -89,7 +90,10 @@ namespace ChimeraTK {
     ChimeraTK::NDRegisterAccessor<UserType>::buffer_2D.resize(1);
     ChimeraTK::NDRegisterAccessor<UserType>::buffer_2D[0].resize(numberOfElements);
 
-    this->_readQueue = cppext::future_queue<void>(3);
+    if(_withReturn) {
+      this->_readQueue = cppext::future_queue<void>(3);
+      this->_accessModeFlags = {AccessMode::wait_for_new_data};
+    }
```

Comment:
I am missing a bit of context what the real consequences of the change are, and how it behaved before.
 * The _readQueue is not always set, which is OK because the check in `doPreRead()`
 * `wait_for_new_data` is set if `_withReturn` is on, which was not the case before. If I understand correctly this is the main functional change of this commit. But I am lacking the overview where it is used, and how it was set before (when adding the slave?)